Makefile

```diff
@@ -22,43 +22,164 @@
+	@echo 'Usage: make <OPTIONS> ... <TARGETS>'
+	@echo ''
+	@echo 'Available targets are:'
+	@echo ''
+	@echo '    help               	Show this help screen.'
+	@echo '    clean               	Remove build files.'
+	@echo
+	@echo '    image                Build and publish the Swagger testing image.'
+	@echo '    rpm                	Build a YUM/SUSE RPM.'
+	@echo '    all 					Build all production artifacts.'
+	@echo
+	@echo '    prepare              Prepare for making an RPM.'
+	@echo '    rpm_build            Builds the RPM.'
+	@echo '    rpm_build_source		Builds the SRPM.'
+	@echo '    rpm_package_source   Creates the RPM source tarball.'
+	@echo
+	@echo '    image_login   		Logs into the Docker registry for pulling and publishing images.'
+	@echo '    image_build   		Builds the Swagger testing image.'
+	@echo '    image_publish   		Builds and publishes the testing image.'
+	@echo ''
+
+clean:
+	rm -rf build dist
 
-rpm: prepare rpm_package_source rpm_build_source rpm_build
+#############################################################################
+# RPM targets
+#############################################################################
+
+SPEC_FILE := ${NAME}.spec
+SOURCE_NAME := ${NAME}-${VERSION}
+
+BUILD_DIR ?= $(PWD)/dist/rpmbuild
+SOURCE_PATH := ${BUILD_DIR}/SOURCES/${SOURCE_NAME}.tar.bz2
+
+rpm: rpm_package_source rpm_build_source rpm_build
 
 prepare:
+	@echo $(NAME)
 	rm -rf $(BUILD_DIR)
 	mkdir -p $(BUILD_DIR)/SPECS $(BUILD_DIR)/SOURCES
 	cp $(SPEC_FILE) $(BUILD_DIR)/SPECS/
 
+# touch the archive before creating it to prevent 'tar: .: file changed as we read it' errors
 rpm_package_source:
-	tar --transform 'flags=r;s,^,/${NAME}-${VERSION}/,' --exclude .git --exclude .github --exclude .gitignore --exclude dist -cvjf $(SOURCE_PATH) .
+	touch $(SOURCE_PATH)
+	tar --transform 'flags=r;s,^,/$(SOURCE_NAME)/,' --exclude .nox --exclude dist/rpmbuild --exclude ${SOURCE_NAME}.tar.bz2 -cvjf $(SOURCE_PATH) .
 
 rpm_build_source:
-	NAME=${NAME} VERSION=${VERSION} rpmbuild -ts $(SOURCE_PATH) --define "_topdir $(BUILD_DIR)"
+	rpmbuild -bs $(BUILD_DIR)/SPECS/$(SPEC_FILE) --target ${ARCH} --define "_topdir $(BUILD_DIR)"
 
 rpm_build:
-	NAME=${NAME} VERSION=${VERSION} rpmbuild -ba $(SPEC_FILE) --define "_topdir $(BUILD_DIR)"
+	rpmbuild -ba $(BUILD_DIR)/SPECS/$(SPEC_FILE) --target ${ARCH} --define "_topdir $(BUILD_DIR)"
 
-venv:
-	${PYTHON} -m venv ${VENV}
-	. ${VENV}/bin/activate; pip install -r requirements.txt
+#############################################################################
+# RPM targets
+#############################################################################
 
-clean:
-	rm -rf ${DIST_DIR}
\ No newline at end of file
+image_login:
+	docker login artifactory.algol60.net
+
+image_build:
+	docker build -t artifactory.algol60.net/csm-docker/stable/craycli/swagger2openapi:latest utils/
+
+image_publish: image_build
+	docker push artifactory.algol60.net/csm-docker/stable/craycli/swagger2openapi:latest
+
+image: image_publish
```

Comment:
These look like they were copied verbatim from `craycli` and they do not apply here.